distribution/openrct2.d.ts

```diff
@@ -2643,6 +2643,11 @@ declare global {
          * @param value Whether to set or clear the flag.
          */
         setFlag(key: PeepFlags, value: boolean): void;
+
+        /**
+         * Gets an array of sprite ids representing the peep's walking animation.
+         */
+        getWalkingSpriteIds(): number[];
```

Comment:
What about combining this with `availableAnimations` ? E.g.
```ts
getAvailableAnimations(): GuestAnimation[];
```
```ts
interface GuestAnimation {
    animation: GuestAnimationTypes;
    sprites: AnimationSprite[];
}

interface AnimationSprite {
    id: number;
    length: number;
}
```
This way it's also possible to get the length of each animation without having to assign it to a peep first. It's also not limited to walking animations either.

Reply:
The use case here is to get the walking animation for a particular peep, whatever their type, so not just guests. The goal is to deliberately make it easier for plugin authors to request the sprite image ids for a particular peep to imitate the guest/staff windows, as requested by @Manticore-007.

To be honest, I'm a bit apprehensive about extending the scope too much. I see where you're coming from, though, and appreciate the suggestion.

Reply:
I'm trying to avoid the situation where the next feature request is to want a different animation sprite id, and then we'd have to add more methods for practically the same thing to the Peep interface. Or deprecate `getWalkingSpriteIds` again to replace it with something else, which is a pain without breaking stuff, as you know.

All added API's should be future proof as much as possible IMHO.

Reply:
Hmm. I wouldn't mind changing it to `getAnimationSpriteIds(animation)`, which would allow the same result with `getAnimationSpriteIds("walking")`. I would prefer to keep a method, rather than a static function, though.

Incidentally, the rotation is currently hardcoded to `1`. I could expose that, but I'd rather not add another untyped number. It's easy enough to add an extra parameter if needed in the future. Hopefully we can agree on that.

Reply:
Fine by me. 🙂 Rotation-wise, I was under the impression it was already returning all rotations because of the array, but I guess that wasn't the case? In the plugin API it already uses rotation indices 0-3 in other places as well though (there's a union for it too).

Reply:
Yes, I was already filtering based on the rotation I mentioned. If there's precedent for passing a rotation parameter, I'll just go with that, though. Thanks.

Reply:
The struct you're referring to doesn't directly map to an internal one, unfortunately. Moreover, I can't guarantee the frames will be loaded sequentially in case of custom objects — especially when sprite maps are used.

Reply:
I've introduced `getAnimationSpriteIds` as discussed above. The implementation differs slightly for Guest and Staff, but the interface is the same.